contracts-periphery/test/Gas.t.sol

```diff
@@ -0,0 +1,423 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "solmate/test/utils/mocks/MockERC20.sol";
+import "test/utils/DSTestPlus.sol";
+import "src/UmbraBatchSend.sol";
+
+interface UmbraToll {
+  function toll() external returns(uint256);
+}
+
+abstract contract UmbraBatchSendGasTest is DSTestPlus {
+  UmbraBatchSend router;
+  MockERC20 token;
+
+  address constant umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
+  address alice = address(0x202204);
+  address bob = address(0x202205);
+
+  uint256 ethBalance;
+  uint256 tokenBalance;
+
+  uint256 toll;
+  bytes32 pkx = "pkx";
+  bytes32 ciphertext = "ciphertext";
+
+  UmbraBatchSend.SendEth[] sendEth;
+  UmbraBatchSend.SendToken[] sendToken;
+
+
+  // Setup params for each function
+  enum Send {ETH, TOKEN, BOTH}
+  address payable[] addrs;
+  uint256 valueAmount; // {value: valueAmount} when calling Umbra
+  uint256 numOfAddrs;
+  uint256 etherAmount; // Ether amount to be sent per address
+  uint256 tokenAmount; // Token amount to be sent per address
+
+  function setUp() virtual public {
+    // Deploy Umbra at an arbitrary address, then place the resulting bytecode at the same address as the production deploys.
+    vm.etch(umbra, (deployCode("test/utils/Umbra.json", bytes(abi.encode(0, address(this), address(this))))).code);
+    router = new UmbraBatchSend(IUmbra(address(umbra)));
+    token = new MockERC20("Test","TT", 18);
+    token.mint(address(this), 1e7 ether);
+    deal(address(this), 1e5 ether);
+
+    ethBalance = address(this).balance;
+    tokenBalance = token.balanceOf(address(this));
+  }
+
+  function testPostSetupState() public {
+    uint256 currentToll = UmbraToll(umbra).toll();
+    assertEq(toll, currentToll);
+  }
+
+  function addParams(Send _type, uint256 numOfAddrs, uint256 etherAmount, uint256 tokenAmount) public {
+    token.approve(address(router), tokenBalance);
```

Comment:
LGTM. However, there's lots of unrelated changes all in the same commit. This removes the benefit of posting the commit hash here because you have to review the full diff anyway. I'd suggest using smaller commits grouped logically. Not every individual comment needs a commit, but you could do e.g. one commit for all chores (whitespace cleanup, comment fixes, etc.), another commit for all refactors, etc.

Leaving this unresolved and will let you resolve once you see this